src/Task.cpp

```diff
@@ -301,12 +303,25 @@ Task::status Task::getStatus () const
   if (! has ("status"))
     return Task::pending;
 
-  return textToStatus (get ("status"));
+  auto status = textToStatus (get ("status"));
+
+  // implement the "virtual" Task::waiting status, which is not stored on-disk
+  // but is defined as a pending task with a `wait` attribute in the future.
+  if (status == Task::pending && is_waiting ()) {
+      return Task::waiting;
+  }
```

Comment:
Should this have a deprecation comment on iit?  (`2.6.0; remove this support in 3.0.0` or something)

Reply:
Good point. Added in https://github.com/GothenburgBitFactory/taskwarrior/pull/2563/commits/00682326526bb54139472007991931a93313a452